src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -19,31 +19,71 @@
 
 package org.elasticsearch.script;
 
-import java.util.Map;
-
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.logging.support.LoggerMessageFormat;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.script.ScriptService.ScriptType;
 
-import static org.elasticsearch.script.ScriptService.ScriptType;
+import java.io.IOException;
+import java.util.Map;
 
 /**
  * Script holds all the parameters necessary to compile or find in cache and then execute a script.
  */
-public class Script {
+public class Script implements ToXContent, Streamable {
+
+    private static final ScriptParser PARSER = new ScriptParser();
+
+    private String script;
+    private ScriptType type;
```

Comment:
if we do make it nullable @colings86 we have to make sure some value gets provided as part of the `canExecuteScript` call, where the type is required to decide whether the script can be executed or not.